components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -511,4 +515,45 @@ public static boolean validateParams(OAuthMessage oAuthMessage, MultivaluedMap<S
         }
         return true;
     }
+
+    /**
+     * This method will start a super tenant flow
+     */
+    public static void startSuperTenantFlow() {
+
+        PrivilegedCarbonContext.startTenantFlow();
+        PrivilegedCarbonContext carbonContext = PrivilegedCarbonContext.getThreadLocalCarbonContext();
+        carbonContext.setTenantId(MultitenantConstants.SUPER_TENANT_ID);
+        carbonContext.setTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+    }
+
+    /**
+     * This API validate the oauth application. Check whether an application exits for given cosumerKey and check
+     * it's status
+     * @param consumerKey clientId
+     * @throws InvalidApplicationClientException
+     */
+    public static void validateOauthApplication(String consumerKey) throws InvalidApplicationClientException {
+
+        String appState = EndpointUtil.getOAuth2Service().getOauthApplicationState(consumerKey);
+
+        if (StringUtils.isEmpty(appState)) {
+            if (log.isDebugEnabled()) {
+                log.debug("A valid OAuth client could not be found for client_id: " + consumerKey);
+            }
+
+            throw new InvalidApplicationClientException("A valid OAuth client could not be found for client_id: " + consumerKey);
+        }
+
+        if (!APP_STATE_ACTIVE.equalsIgnoreCase(appState)) {
```

Comment:
`!APP_STATE_ACTIVE.equalsIgnoreCase(appState)` shall we extract this to a new method to improve readability